src/viam/sdk/CMakeLists.txt

```diff
@@ -142,6 +142,8 @@ target_sources(viamsdk
       ../../viam/sdk/common/exception.hpp
       ../../viam/sdk/common/linear_algebra.hpp
       ../../viam/sdk/common/pose.hpp
+      ../../viam/sdk/common/proto_convert.hpp
+      ../../viam/sdk/common/proto_convert_vector.hpp
```

Comment:
⬆️  - How close do the changes in this review get things w.r.t. removing this:
```
# This bit of magic is required so that we get the public build-time
# include path of viamapi in *ahead* of the path anchored to `src` by
# `BASE_DIRS` below, because otherwise we will pull in the static
# versions of the proto files even if we are generating dynamically.
target_include_directories(viamsdk
  PUBLIC
    "$<BUILD_INTERFACE:$<TARGET_PROPERTY:viam-cpp-sdk::viamapi,INTERFACE_INCLUDE_DIRECTORIES>>"
)
```

Or at least reducing it to `PRIVATE`?